src/main/java/Parser.java

```diff
@@ -0,0 +1,174 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+/**
+ * Represents the parser of the Duke application. The parser is responsible for
+ * analysing the user command so that it can be understood by the application.
+ */
+public class Parser {
+
+    /**
+     * Parses the dates/times of deadlines and events.
+     * @param date String that represents either a date in YYYY-MM-DD
+     *             format, or a timestamp in YYYY-MM-DD HHMM format.
+     * @return String that represents an alternative format of the date/time.
+     * @throws DateException If the given date/time is not in a format
+     * appropriate for parsing.
+     */
+    public static String parseDate(String date) throws DateException {
+        try {
+            String[] dates = date.split(" ");
+            String processedDate = "";
+            String processedTime = "";
+            if (dates.length == 1) {
+                LocalDate localDate = LocalDate.parse(dates[0]);
+                processedDate = localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+                return processedDate;
+            } else {
+                LocalDate localDate = LocalDate.parse(dates[0]);
+                processedDate = localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+
+                double time = Double.parseDouble(dates[1]);
+                if (time >= 1300) {
+                    time -= 1200;
+                }
+                int hour = (int) (time / 100);
+                if (hour == 24 || hour == 0) {
+                    processedTime += "12.";
+                } else {
+                    processedTime += hour + ".";
+                }
+                int minute = (int) (time % 100);
+                if (minute < 10) {
+                    processedTime += "0" + minute;
+                } else {
+                    processedTime += minute;
+                }
+                if ((Double.parseDouble(dates[1]) >= 1200)
+                        && (Double.parseDouble(dates[1]) < 2400)) {
+                    processedTime += "pm";
+                } else {
+                    processedTime += "am";
+                }
```

Comment:
Instead of manually processing the date/time, you can use built-in Java libraries to do so.